kong/utils.go

```diff
@@ -283,7 +311,49 @@ func fillConfigRecord(schema gjson.Result, config Configuration) Configuration {
 //	 ...
 //	}
 func flattenDefaultsSchema(schema gjson.Result) Schema {
-	value := schema.Get("fields")
+	fields := schema.Get("fields")
+	if fields.Exists() {
+		return flattenLuaSchema(fields)
+	}
+	properties := schema.Get("properties")
+	if properties.Exists() {
+		return flattenJSONSchema(properties)
+	}
+	return Schema{}
+}
+
+func flattenJSONSchema(value gjson.Result) Schema {
+	results := Schema{}
+
+	value.ForEach(func(key, value gjson.Result) bool {
+		name := key.String()
+
+		ftype := value.Get("type")
+		// when type==object and additionalProperties==false, the object
+		// represents either a foreign relationship or a map entry.
+		// In both cases, defaults don't need to be injected.
+		additionalProperties := value.Get("additionalProperties")
+		if ftype.String() == "object" &&
+			(!additionalProperties.Exists() ||
+				(additionalProperties.Exists() &&
+					additionalProperties.Bool())) {
```

Comment:
Shouldn't this be `!additionalProperties.Bool()` ? 🤔

Reply:
Nope, we don't want to execute the following logic when `additionalProperties` is `false`